cmake/VASTGitVersion.cmake

```diff
@@ -31,3 +33,36 @@ string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.[0-9]+\\-([0-9]+).*" "\\1"
                      VAST_VERSION_TWEAK "${VAST_VERSION_TAG}")
 string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.[0-9]+\\-[0-9]+\\-(.*)" "\\1"
                      VAST_VERSION_COMMIT "${VAST_VERSION_TAG}")
+
+# Strip - and -0- suffixes from the version tag.
+string(REGEX REPLACE "^(.*)[-0-|-]\\\$" "\\1" VAST_VERSION_TAG
+                     "${VAST_VERSION_TAG}")
+
+file(
+  GLOB_RECURSE
+  hash_files
+  "${CMAKE_CURRENT_LIST_DIR}/../libvast/*.hpp"
+  "${CMAKE_CURRENT_LIST_DIR}/../libvast/*.hpp.in"
+  "${CMAKE_CURRENT_LIST_DIR}/../libvast/*.cpp"
+  "${CMAKE_CURRENT_LIST_DIR}/../cmake/*"
+  "${CMAKE_CURRENT_LIST_DIR}/../CMakeLists.txt")
```

Comment:
Why has the hash computation been moved to this file?
The way this is written it would break if this file is included from the top level CMakeLists.txt, or any other cmake file that is not in the right relative location.

Reply:
It had to be moved to we recompute it alongside the version.

> The way this is written it would break if this file is included from the top level CMakeLists.txt, or any other cmake file that is not in the right relative location.

That is not true, we use `CMAKE_CURRENT_LIST_DIR` over `CMAKE_CURRENT_SOURCE_DIR` here to avoid that. It's relative to the location of the `VASTGitVersion.cmake` file this way.

Reply:
Also, fun fact: this already is being included from both the top-level CMakeLists.txt and from a generated .cmake file in some binary directory.